pandas/tests/indexes/multi/test_indexing.py

```diff
@@ -183,6 +183,155 @@ def test_get_indexer():
         idx1.get_indexer(idx2)
 
 
+def test_get_indexer_and_fill():
```

Comment:
in all of these test cases, they illustrate the behavior up until 0.23.0.  The assumption of this PR is that, in the absence of documentation about these changes to MultiIndex behavior in 0.23.0 (and that this PR does not break any existing tests), that those changes are bugs